localstack/aws/protocol/serializer.py

```diff
@@ -708,29 +709,88 @@ def _serialize_error(
         error: ServiceException,
         code: str,
         sender_fault: bool,
-        serialized: HttpResponse,
+        response: HttpResponse,
         shape: Shape,
         operation_model: OperationModel,
     ) -> None:
         # TODO handle error shapes with members
         body = {"__type": code, "message": str(error)}
-        serialized.set_json(body)
+        response.set_json(body)
 
-    def _serialize_payload(
+    def _serialize_response(
         self,
         parameters: dict,
-        serialized: HttpResponse,
+        response: HttpResponse,
         shape: Optional[Shape],
         shape_members: dict,
         operation_model: OperationModel,
     ) -> None:
         json_version = operation_model.metadata.get("jsonVersion")
         if json_version is not None:
-            serialized.headers["Content-Type"] = "application/x-amz-json-%s" % json_version
+            response.headers["Content-Type"] = "application/x-amz-json-%s" % json_version
+
+        if shape is None:
+            response.set_json({})
+            return
+
+        # first, serialize everything into a dictionary
+        data = {}
+        self._serialize(data, parameters, shape)
+
+        if shape.serialization is None:
+            # if there are no special serialization instructions then just set the data dict as JSON payload
+            response.set_json(data)
+            return
+
+        # otherwise, move data attributes into their appropriate locations
+        # (some shapes have a serialization dict, but don't actually have
```

Comment:
I think one line got lost here.
Some shape have a serialization dict, but don't actually have... additional metadata in there?

Reply:
ah yea got distracted mid-sentence :D they have a serialization dict but don't actually need special serialization. :shrug: